### PR TITLE
Swapping vscode calls for ApiWrapper for testability

### PR DESCRIPTION
**Pull request by Benjin**
Also adds tests to cover controller calls with user input for negative cases

### CODE REVIEW

File: extensions/sql-database-projects/src/controllers/mainController.ts [hunk 1]
```diff
@@ -3,36 +3,39 @@
...
 
 const SQL_DATABASE_PROJECTS_VIEW_ID = 'sqlDatabaseProjectsView';
 
 /**
  * The main controller class that initializes the extension
  */
-export default class MainController implements vscode.Disposable {
-	protected _context: vscode.ExtensionContext;
+export default class MainController implements Disposable {
+	//protected _context: vscode.ExtensionContext;
```
Comment (kisantia) [inline extensions/sql-database-projects/src/controllers/mainController.ts:26]: should be removed

File: extensions/sql-database-projects/src/controllers/mainController.ts [hunk 2]
```diff
@@ -3,36 +3,39 @@
...
 	protected dbProjectTreeViewProvider: SqlDatabaseProjectTreeViewProvider = new SqlDatabaseProjectTreeViewProvider();
 	protected projectsController: ProjectsController;
 	protected netcoreTool: NetCoreTool;
 
-	public constructor(context: vscode.ExtensionContext) {
-		this._context = context;
-		this.projectsController = new ProjectsController(this.dbProjectTreeViewProvider);
+	public constructor(private context: ExtensionContext, private apiWrapper: ApiWrapper) {
+		//this._context = context;
```
Comment (kisantia) [inline extensions/sql-database-projects/src/controllers/mainController.ts:32]: also should be removed
